src/render-instruction.ts

```diff
@@ -116,13 +117,38 @@ ${typeMapperImports.join('\n')}`.trim()
   // Accounts
   // -----------------
   private processIxAccounts(): ProcessedAccountKey[] {
-    return this.ix.accounts.map((acc) => {
-      const knownPubkey = resolveKnownPubkey(acc.name)
-      const optional = acc.optional ?? false
-      return knownPubkey == null
-        ? { ...acc, optional }
-        : { ...acc, knownPubkey, optional }
+    let processedAccountsKey: ProcessedAccountKey[] = []
+    this.ix.accounts.map((acc: IdlInstructionAccount | IdlAccountsCollection) => {
+      if(this.isAccountsCollection(acc)){
+        acc.accounts.map((ac) => {
+          // as there are cases where the collection of the accounts is reused on the same ix, is needed to change the name of the account
+          ac.name += acc.name.charAt(0).toUpperCase().concat(acc.name.slice(1))
+          const knownPubkey = resolveKnownPubkey(ac.name)
+          const optional = ac.optional ?? false
+          if (knownPubkey == null) {
+            processedAccountsKey.push({ ...ac, optional })
+          }
+          else{
+            processedAccountsKey.push({ ...ac, knownPubkey, optional })
+          }
+        })
+      }
+      else {
```

Comment:
Please follow code standards here and/or run `yarn lint:fix` (there is a `.prettierrc` in the root which your editor doesn't seem to respect).
The lint part of the workflow most likely will fail due to this.